libs/clj-icat-direct/src/clj_icat_direct/queries.clj

```diff
@@ -565,28 +560,28 @@
            MAX(a.access_type_id)                  as access_type_id
       FROM r_coll_main c
       JOIN r_objt_access a ON c.coll_id = a.object_id
-      JOIN parent p ON c.parent_coll_name = p.coll_name
       JOIN r_objt_metamap mm ON mm.object_id = c.coll_id
       JOIN r_meta_main m ON m.meta_id = mm.meta_id 
      WHERE a.user_id IN ( SELECT group_user_id FROM user_groups )
        AND c.coll_type != 'linkPoint'
+       AND c.parent_coll_name = ?
        AND m.meta_attr_name = 'ipc_UUID'
   GROUP BY dir_name, full_path, base_name, c.create_ts, c.modify_ts, type, data_size, uuid
   ORDER BY base_name ASC"
 
    :count-files-in-folder
-   "WITH user_groups AS ( SELECT g.*
+   "WITH user_groups AS ( SELECT g.group_user_id
                             FROM r_user_main u
                             JOIN r_user_group g ON g.user_id = u.user_id
                            WHERE u.user_name = ?
                              AND u.zone_name = ? ),
 
-         parent      AS ( SELECT * from r_coll_main
+         parent      AS ( SELECT coll_id from r_coll_main
                            WHERE coll_name = ? ),
 
-         data_objs   AS ( SELECT *
+         data_objs   AS ( SELECT data_id
                             FROM r_data_main
-                           WHERE coll_id = ANY(ARRAY( SELECT coll_id FROM parent )))
+                           WHERE coll_id IN ( SELECT coll_id FROM parent ))
```

Comment:
If it is, it's not enforced:

```
ICAT=# \d r_coll_main
                         Table "public.r_coll_main"
      Column      |          Type           |           Modifiers            
------------------+-------------------------+--------------------------------
 coll_id          | bigint                  | not null
 parent_coll_name | character varying(2700) | not null
 coll_name        | character varying(2700) | not null
 coll_owner_name  | character varying(250)  | not null
 coll_owner_zone  | character varying(250)  | not null
 coll_map_id      | bigint                  | default 0
 coll_inheritance | character varying(1000) | 
 coll_type        | character varying(250)  | default '0'::character varying
 coll_info1       | character varying(2700) | default '0'::character varying
 coll_info2       | character varying(2700) | default '0'::character varying
 coll_expiry_ts   | character varying(32)   | 
 r_comment        | character varying(1000) | 
 create_ts        | character varying(32)   | 
 modify_ts        | character varying(32)   | 
Indexes:
    "idx_coll_main2" UNIQUE, btree (parent_coll_name, coll_name)
    "idx_coll_main3" UNIQUE, btree (coll_name)
    "idx_coll_main1" btree (coll_id)
    "idx_coll_main_coll_type" btree (coll_type)
    "idx_coll_main_parent_coll_name" btree (parent_coll_name)

```

Only `coll_name` and `parent_coll_name, coll_name` are made unique.